packages/go_router/lib/src/builder.dart

```diff
@@ -151,18 +104,19 @@ class GoRouterDelegate extends RouterDelegate<List<GoRouteMatch>>
 
     // pass either the match error or the build error along to the navigator
     // builder, preferring the match error
-    if (matches.length == 1 && matches.first.error != null) {
-      error = matches.first.error;
+    if (matches.matches.length == 1 && matches.matches.first.error != null) {
+      error = matches.matches.first.error;
     }
 
     // wrap the returned Navigator to enable GoRouter.of(context).go()
     final Uri uri = Uri.parse(location);
     return builderWithNav(
       context,
       GoRouterState(
-        _parser,
+        configuration,
         location: location,
-        name: null, // no name available at the top level
+        name: null,
+        // no name available at the top level
```

Comment:
What do you think of moving comments to before the argument? Same comment for line 130.

```suggestion
        // no name available at the top level
        name: null,
```